app/signals/apps/media/views.py

```diff
@@ -0,0 +1,46 @@
+# SPDX-License-Identifier: MPL-2.0
+# Copyright (C) 2024 Delta10 B.V.
+import mimetypes
+import os
+from datetime import timedelta
+
+from django.conf import settings
+from django.core import signing
+from django.http import HttpRequest, HttpResponse
+from django.http.response import HttpResponseBase
+from django.views.static import serve
+
+signer = signing.TimestampSigner(salt='protected_file_system_storage')
+
+
+def download_file(request: HttpRequest, path: str) -> HttpResponseBase:
+    t = request.GET.get('t')
+    s = request.GET.get('s')
+
+    if settings.DEBUG:
+        return serve(request, path, document_root=settings.MEDIA_ROOT, show_indexes=False)
+
+    if not t or not s:
+        return HttpResponse('No signature provided', status=401)
+
+    try:
+        signer.unsign(f'{path}:{t}:{s}', max_age=timedelta(hours=1))
```

Comment:
Could you make this timedelta value configurable through the settings?

Reply:
@4c0n forgot about this PR. Just added a commit to make this configurable.